src/Entity/Int32EntityId.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Wikibase\DataModel\Entity;
+
+use InvalidArgumentException;
+
+/**
+ * Interface for EntityIds that can be converted to a unique positive 32 bit integer in the range
+ * [1..2147483647], and back from the entity type and the number. Entity types that do not meet this
+ * criteria should not implement this interface.
+ *
+ * Entity types are not required and not guaranteed to implement this interface. Use the full string
+ * serialization whenever you can and avoid using numeric IDs.
+ *
+ * @since 6.1
+ *
+ * @license GPL-2.0+
+ * @author Thiemo Mättig
+ */
+interface Int32EntityId {
+
+	/**
+	 * @since 6.1
+	 *
+	 * @param string $idSerialization
+	 *
+	 * @throws InvalidArgumentException When the provided serialization does not allow getNumericId
+	 *  to return a unique positive 32 bit integer.
+	 */
+	public function __construct( $idSerialization );
+
+	/**
+	 * @since 6.1
+	 *
+	 * @return int Guaranteed to be a unique positive 32 bit integer.
+	 */
+	public function getNumericId();
+
+	/**
+	 * @since 6.1
+	 *
+	 * @param int|float|string $numericId
```

Comment:
From reading "allow getNumericId to return a unique integer in the range...", it's not at all clear to me what the input "010" would to. Would that raise an error? Or would it ignore the leading "0"? The integer that is returned is unique in both cases (actually, all integers are unique). 
We could try to say that all different inputs to this method must result in a different integer being returned from getNumericId. But that's not strictly true: 0x10 and '16' and 16.0 all yield the same value for getNumericId. Why shouldn't '016' work, too? It seems natural that it would. 
The documentation seems unclear about this point: should '016' work, or trigger an exception? I think it would be good to clarify it.